src/mcts/search.cc

```diff
@@ -956,7 +956,7 @@ SearchWorker::NodeToProcess SearchWorker::PickNodeToExtend(
         ++possible_moves;
       }
       const float Q = child.GetQ(fpu);
-      const float score = child.GetU(puct_mult) + Q;
+      const float score = child.GetU(puct_mult) * (2*Q/log((1 + Q)/(1 - Q))) + Q;
```

Comment:
Mathematically, the formula is correct and gives a correct continuous function in [-1,1] if you consider the limits for Q going to -1, 0 and +1.

Translating the formula directly into C++ leads to trouble:
- for Q = 0, we get 0/0 (NaN).
- for Q = -1, we get log(0), which is undefined according to http://www.cplusplus.com/reference/cmath/log/ (may lead to a pole error depending on library implementation).

These numerical aspects aside, you may consider a faster log from the `fastmath.h` which will be considerably faster and will lead to an nps boost on fast backends.

Reply:
For now at least, the speed doesn't matter too much. The numerical issues do though

Reply:
The undefined values did occur to me, but I'm not very fluent in C++, so if anyone wants to propose an alternative formulation to resolve them, it would be much appreciated.

Reply:
For now, 2 if statements is probably the easiest